syntax = "proto3";

package tunechain;

import "google/protobuf/empty.proto";

service MusicService {
  rpc RegisterMusic (MusicRequest) returns (MusicResponse);
  rpc ListMusics (google.protobuf.Empty) returns (stream MusicData);
  rpc PlayMusic (PlayRequest) returns (MusicResponse);
}

message MusicRequest {
  string title = 1;
  string artist = 2;
  string producer = 3;
  int32 revenue_split = 4;
}

message PlayRequest {
  string title = 1;
  int32 times = 2;
}

message MusicResponse {
  bool success = 1;
  string message = 2;
}

message MusicData {
  string title = 1;
  string artist = 2;
  string producer = 3;
  int32 revenue_split = 4;
  int32 plays = 5;
  double revenue = 6;
}
